virtcontainers/qemu.go

```diff
@@ -931,6 +931,17 @@ func (q *qemu) waitSandbox(timeout int) error {
 	return nil
 }
 
+func isConnectionRefusedError(err error) bool {
+	if neterr, ok := err.(*net.OpError); ok {
+		if syserr, ok := neterr.Err.(*os.SyscallError); ok {
+			if syserr.Error() == "connect: connection refused" {
```

Comment:
Wow, they don't make it easy do they? :smile: `SyscallError` is an odd beast - why we cannot check the `errno` value directly I have no idea. What worries me is that the design of `SyscallError` looks rather fragile to me: if that package suddenly supported i18n, this would break. But I've found a solution:

```go
if errno, ok := syserr.Err.(syscall.Errno); ok {
    if errno == unix.ECONNREFUSED {
        return true
    }
}
```

Reply:
>  What worries me is that the design of `SyscallError` looks rather fragile to me: if that package suddenly supported i18n, this would break.

Yeah, true. I didn't particularly like comparing the error string either :-/

> But I've found a solution

Cool, thanks. I'll rework that.